src/js/containers/Submission.jsx

```diff
@@ -16,11 +16,46 @@ import Signature from './Signature.jsx'
 import Summary from './Summary.jsx'
 import RefileButton from '../containers/RefileButton.jsx'
 import ParseErrors from './ParseErrors.jsx'
+import LoadingIcon from '../components/LoadingIcon.jsx'
 
 const EditsNav = submissionProgressHOC(EditsNavComponent)
 const NavButton = submissionProgressHOC(NavButtonComponent)
 const RefileWarning = submissionProgressHOC(RefileWarningComponent)
 
+const renderByCode = (code, page, message) => {
+  const toRender = []
+  if(code === -1) {
+    toRender.push(<p>{message}</p>)
+  }else{
+    if(page === 'upload'){
+      toRender.push(<UploadForm code={code}/>)
+      if(code === 5) {
+        toRender.push(<RefileWarning/>)
+        toRender.push(<ParseErrors/>)
+      }
```

Comment:
For the `/upload` page, I think we should add `RefileWarning` if the code is 8 too. Seems like it would be good to let them know on all pages.